src/components/main/menu/NCSelector.jsx

```diff
@@ -0,0 +1,170 @@
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import propTypes from 'proptypes';
+
+import { updateNC } from '../../../redux/reducers/data';
+
+import { COUNCILS } from '../../common/CONSTANTS';
+import Checkbox from '../../common/Checkbox';
+
+const NCSelector = ({
+  updateNCList,
+}) => {
+  const [searchValue, setSearchValue] = useState('');
+  const [filteredCouncilList, setFilteredCouncilList] = useState(COUNCILS);
+  const [selectedCouncilList, setSelectedCouncilList] = useState(
+    COUNCILS.reduce((acc, council) => {
+      acc[council] = false;
+      return acc;
+    }, { all: false }),
+  );
+
+  const selectRowStyle = {
+    margin: '0 0 7px 0',
+  };
+
+  const selectRowTextStyle = {
+    width: '280px',
+  };
+
+  const handleSearch = (e) => {
+    const term = e.target.value;
+    const searchFilter = new RegExp(term, 'i');
+    const searchList = COUNCILS.filter((council) => searchFilter.test(council));
+    setFilteredCouncilList(searchList);
+    setSearchValue(e.target.value);
+  };
+
+  const handleSelectCouncil = (council) => {
+    const newSelectedCouncilList = { ...selectedCouncilList };
+
+    switch (council) {
+      case 'all': {
+        let value = true;
+
+        if (newSelectedCouncilList.all) {
+          newSelectedCouncilList.all = false;
+          value = false;
+        }
+
+        Object.keys(newSelectedCouncilList).forEach((c) => {
+          newSelectedCouncilList[c] = value;
+        });
+        break;
+      }
+      default:
+        newSelectedCouncilList.all = false;
+        newSelectedCouncilList[council] = !newSelectedCouncilList[council];
+        break;
+    }
+
+    const newNCList = Object.keys(newSelectedCouncilList).filter((c) => newSelectedCouncilList[c] && c !== 'all');
+
+    setSelectedCouncilList(newSelectedCouncilList);
+    updateNCList(newNCList);
```

Comment:
Is this better than just storing all the neighborhood councils in redux state as booleans?

Reply:
My thinking was that the store doesn't care about managing check boxes, it just needs a list of all councils to reference it across the application. What do you think?

Reply:
My thought process was that it seemed like a number of potentially unnecessary tasks on each council select/deselect. On every select it's:
1. Creating a copy of the existing council state 
2. Looping through the councils or updating the selected council in the copy
3. Looping again to build newNCList
4. Updating local council state which will run React's diff algorithm to determine re-render
5. Updating redux council state

If we store the councils in Redux as booleans and map state to props here, React will handle the entire diff operation (on props change) and you won't have to manually rebuild state in the click handler. The `updateNC` action might be replaced with three others that: update single council, select all councils, deselect all councils. Update single council would just flip the boolean in state for the corresponding council. For select/deselect all, we could have prebuilt states (all false/true) that replace the existing council state in redux.

Then we have another action tied to the overall Submit button that loops through councils in redux state to extract all with `true` values to send to the back end. This way we only have to do one loop through councils when we are ready to request data from the server. 

I'm going to merge this in but maybe we can discuss redux at tomorrow's meeting.